onnxruntime/core/framework/session_state_flatbuffers_utils.cc

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/session_state_flatbuffers_utils.h"
+
+namespace onnxruntime::experimental::utils {
+
+std::string GetSubgraphId(const NodeIndex node_idx, const std::string& attr_name) {
+  return std::to_string(node_idx) + "_" + attr_name;
+}
+
+FbsSessionStateViewer::FbsSessionStateViewer(const fbs::SessionState& fbs_session_state)
+    : fbs_session_state_{fbs_session_state} {
+}
+
+Status FbsSessionStateViewer::Validate() const {
+  if (fbs_session_state_.sub_graph_session_states() == nullptr) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "SessionState for subgraphs is null. Invalid ORT format model.");
+  }
+
+  const auto* const fbs_kcis = fbs_session_state_.kernels();
+  if (fbs_kcis == nullptr) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Kernel create info is null. Invalid ORT format model.");
+  }
+
+  const auto* const fbs_node_indices = fbs_kcis->node_indices();
+  if (fbs_node_indices == nullptr) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Kernel create info node indices are null. Invalid ORT format model.");
+  }
+
+  const auto* const fbs_kernel_def_hashes = fbs_kcis->kernel_def_hashes();
+  if (fbs_kernel_def_hashes == nullptr) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Kernel create info hashes are null. Invalid ORT format model.");
+  }
+
+  if (fbs_node_indices->size() != fbs_kernel_def_hashes->size()) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
+                           "Size mismatch for kernel create info node indexes and hashes. Invalid ORT format model.",
+                           fbs_node_indices->size(), " != ", fbs_kernel_def_hashes->size());
+  }
+
```

Comment:
If we need to squeeze size down more we could avoid so many ORT_MAKE_STATUS calls here with a 
```
std::string error;
do {
  if (condition fails) {
    error = ...
    break;
  }
  repeat

} while (false)

return error.empty() ? Status::OK() : ORT_MAKE_STATUS(...)
```

Reply:
FYI, tried it and did not observe a significant decrease in SizeBench